gcc/rust/parse/rust-parse-impl.h

```diff
@@ -4922,8 +4922,8 @@ Parser<ManagedTokenSource>::parse_trait (AST::Visibility vis,
 
   if (is_auto_trait && !trait_items.empty ())
     {
-      add_error (
-	Error (locus, "associated items are forbidden within auto traits"));
+      rust_error_at (locus, ErrorCode::E0380,
+		     "auto traits cannot have methods or associated items");
```

Comment:
yes, this does change the behavior of the code. `add_error` adds the created error to an internal table in the parser, which can choose to emit them later on or not. so we need to keep using `add_error` instead of `rust_error_at`